server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -390,11 +431,17 @@ private HitContext prepareNestedHitContext(SearchContext context,
     }
 
     private void loadStoredFields(Function<String, MappedFieldType> fieldTypeLookup,
+                                  Profiler profileListener,
                                   CheckedBiConsumer<Integer, FieldsVisitor, IOException> fieldReader,
                                   FieldsVisitor fieldVisitor, int docId) throws IOException {
-        fieldVisitor.reset();
-        fieldReader.accept(docId, fieldVisitor);
-        fieldVisitor.postProcess(fieldTypeLookup);
+        try {
```

Comment:
As a heads up, we've brainstormed moving stored fields loading to its own subphase. I'm not sure it will happen, but this would affect the output structure (which is fine, it's not actually part of our API).

Reply:
++ - would we want to do only a single stored field load?

Reply:
Yep I think that'd be the goal.